app/crons/process-on-hold/process.js

```diff
@@ -7,47 +7,29 @@ const getCommonData = () => {
   return { searchType: 'status', searchText: 'ON HOLD', datePast24Hrs: now.setDate(now.getDate() - 1) }
 }
 
-const processOnHoldApplications = async () => {
-  try {
-    const { searchType, searchText, datePast24Hrs } = getCommonData()
-    const apps = await getApplications(searchType, searchText, 9999, 0, [], { field: 'CREATEDAT', direction: 'ASC' })
-
-    if (apps.total > 0) {
-      const applicationRefs = apps.applications.filter(a => new Date(a.updatedAt) <= datePast24Hrs).map(a => a.reference)
-      console.log(`${new Date().toISOString()} processing OnHold applications:processing records: ${JSON.stringify({ applicationRefs })}`)
-      for (const appRef of applicationRefs) {
-        await processApplicationClaim(appRef, 'admin', true)
-        console.log(`${new Date().toISOString()} processing OnHold applications:processing : ${JSON.stringify({ appRef })}`)
-      }
-      return true
-    } else {
-      console.log(`${new Date().toISOString()} processing OnHold applications:Nothing to process today`)
+const processOnHoldApplications = async (logger) => {
+  const { searchType, searchText, datePast24Hrs } = getCommonData()
+  const apps = await getApplications(searchType, searchText, 9999, 0, [], { field: 'CREATEDAT', direction: 'ASC' }, logger)
+  logger.setBindings({ applicationsTotal: apps.total })
+  if (apps.total > 0) {
+    const applicationRefs = apps.applications.filter(a => new Date(a.updatedAt) <= datePast24Hrs).map(a => a.reference)
+    logger.setBindings({ applicationRefs })
+    for (const appRef of applicationRefs) {
```

Comment:
How many applicationRefs can we be looping through here? Just asking because having an awaited function inside a for loop could end up being _**real**_ slow, is there a chance we should be using await Promise.all here? Or something similiar.

If you think its out of scope, which it very reasonably could be, then maybe just needs flagging to revisit.

Reply:
Yeah fair call, it could be a few (however many happen in a day). we could get the refs and promise all or even just remove the await as we expect no return values, it seems to be just fire and forget.

Although as it's a daily cron job, not sure how much the speed of execution really matters.

Reply:
Thats fair yeah - lets not spend time trying to fix something that might not need fixing.